examples/list-datatype.kl

```diff
@@ -0,0 +1,22 @@
+#lang "n-ary-app.kl"
+
+(import "defun.kl")
+
+(datatype (List A)
+  (nil)
+  (:: A (List A)))
+
+(defun map (f lst)
+  (case lst
+    [(:: x xs) (:: (f x) (map f xs))]
+    [(nil) (nil)]))
+
+(defun foldr (f base lst)
+  (case lst
+    [(:: x xs)
+     (f x (foldr f base xs))]
+    [(nil)
+     base]))
```

Comment:
this code makes me very happy, klister finally feels like a real programming language!

Reply:
Yeah, it's getting more and more fun to actually use!